atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java

```diff
@@ -264,16 +268,33 @@ private boolean isHostHealthy(InetSocketAddress host) {
     }
 
     private CassandraClientPoolingContainer getRandomGoodHost() {
+        // Note that this is safe (unless there are 0 hosts in the pool, in which case IllegalStateException is not
+        // unreasonable)
+        return getRandomGoodHostForPredicate(address -> true).get();
```

Comment:
I think having `.orThrow(...)` would be slightly better where we return something saying there are no hosts.